sdks/wasm/Mono.WebAssembly.DebuggerProxy/FirefoxMonoProxy.cs

```diff
@@ -0,0 +1,510 @@
+			if (actor == null)
+				goto ReadAgain;
+
+			SendCommandToBrowser (actor, "prototypeAndProperties", "", Browser, ide);
+		ReadAgain2:
+			bytesRec = Browser.Read (bytes, 0, bytes.Length);
+			//System.Console.WriteLine (Encoding.ASCII.GetString (bytes, 0, bytesRec));
+			callStack = getFirstMessage (Encoding.ASCII.GetString (bytes, 0, bytesRec));
+			call_stack = JObject.Parse (callStack);
+			var mono_frame = call_stack ["ownProperties"]? ["0"]? ["value"]? ["preview"]? ["ownProperties"];
+			if (mono_frame == null)
+				goto ReadAgain2;
+			var il_pos = mono_frame ["il_pos"] ["value"].Value<int> ();
+			var method_token = mono_frame ["method_token"] ["value"].Value<int> ();
+			var assembly_name = mono_frame ["assembly_name"] ["value"].Value<string> ();
+			var framesList = new List<Frame> ();
+			var asm = store.GetAssemblyByName (assembly_name);
+			var method = asm.GetMethodByToken (method_token);
+
+			//TODO: do a loop to read all callstack
+
+			var location = method?.GetLocationByIl (il_pos);
+			int frame_id = 0;
+			framesList.Add (new Frame (method, location, frame_id));
+
+
+			var frames = JObject.FromObject (new {
+				frames = new JArray (),
+				from = from
+			});
+
+			var frame = JObject.FromObject (new {
+				actor = $"dotnet:scope:{frame_id}",
+				type = "call",
+				displayName = method.Name,
+				where = JObject.FromObject (new {
+					actor = location.Id.ToString (),
+					line = location.Line,
+					column = location.Column
+				})
+			});
+			((JArray)frames ["frames"]).Add (frame);
+			++frame_id;
+			current_callstack = framesList;
+			var msg = frames.ToString ();
+			msg = (msg.Length).ToString () + ":" + msg;
+			//System.Console.WriteLine (msg);
+			ide.Write (Encoding.ASCII.GetBytes (msg), 0, msg.Length);
+			readedBrowser = getRestOfMessage (Encoding.ASCII.GetString (bytes, 0, bytesRec));
+		}
+
+		private  void SendBreakpointToMono (JObject details, NetworkStream Browser, NetworkStream ide)
+		{
+			if (store == null)
+				return;
+			string msg = ":{ \"from\":\"" + details ["to"].ToString () + "\"}";
+			msg = (msg.Length - 1).ToString () + msg;
+			//System.Console.WriteLine (msg);
+			ide.Write (Encoding.ASCII.GetBytes (msg), 0, msg.Length);
+
+
+			BreakPointRequest req = BreakPointRequest.ParseFirefox (details, store);
+			var bp_loc = store.FindBestBreakpoint (req);
+			var bp = new Breakpoint (bp_loc, local_breakpoint_id++, BreakPointState.Pending);
+			var asm_name = bp.Location.CliLocation.Method.Assembly.Name;
+			var method_token = bp.Location.CliLocation.Method.Token;
+			var il_offset = bp.Location.CliLocation.Offset;
+
+			var expression = string.Format ("MONO.mono_wasm_set_breakpoint(\"{0}\", {1}, {2})", asm_name, method_token, il_offset);
+			SendCommandToBrowser (actorName, "evaluateJS", expression, Browser, ide);
+		}
+
+		private  void SendBreakpointPositionsCompressed (string script_id, NetworkStream Browser, NetworkStream ide, string line)
```

Comment:
nit: (here and in some other methods) dont' need to write `private` it's the default.